### PR TITLE
Switch to node-fetch

### PR DESCRIPTION
**Pull request by InternetProtocolv6**

### CODE REVIEW

File: src/lib/Client.js [hunk 1]
```diff
@@ -44,6 +44,7 @@ class IdioticClient {
      * @type {String}
      */
     this.baseUrl = options.url || this.dev ? "https://dev.anidiots.guide/" : "https://api.anidiots.guide/api/";
+    this.headers = { [`${this.dev ? "Authorization" : "token"}`]: token };
```
Comment (MrJacz) [inline src/lib/Client.js]: this should be done in `_get`

File: src/lib/Client.js [hunk 2]
```diff
@@ -789,13 +790,16 @@ class IdioticClient {
...
   _get(endpoint, query = {}) {
-    return snekfetch.get(`${this.baseUrl}${endpoint}`)
-      .set(this.dev ? "Authorization" : "token", this.token)
-      .query(query)
-      .then(res => {
-        if (res.status !== 200) throw res;
-        return res.body;
-      });
+    const url = new URL(`${this.baseUrl}${endpoint}`);
+    Object.keys(query).forEach(key => url.searchParams.append(key, query[key]));
```
Comment (MrJacz) [inline src/lib/Client.js]:
```javascript
for (const key of Object.keys(query)) url.searchParams.append(key, query[key]);
```

File: src/lib/Client.js [hunk 3]
```diff
@@ -789,13 +790,16 @@ class IdioticClient {
...
+
+    return fetch(url, 
+      { method: "GET",
+        headers: this.headers,
```
Comment (MrJacz) [inline src/lib/Client.js]:
```js
{ [this.dev ? "Authorization" : "token"]: this.token }
```

File: src/lib/Client.js [hunk 4]
```diff
@@ -789,13 +790,16 @@ class IdioticClient {
...
+      }).then(res => {
+      if (res.status !== 200) throw res;
+      return res.body._outBuffer;
```
Comment (MrJacz) [inline src/lib/Client.js]: use `res.buffer()`

File: src/lib/Client.js [hunk 5]
```diff
@@ -789,13 +790,16 @@ class IdioticClient {
...
-        return res.body;
-      });
+    const url = new URL(`${this.baseUrl}${endpoint}`);
+    Object.keys(query).forEach(key => url.searchParams.append(key, query[key]));
+
+    return fetch(url, 
+      { method: "GET",
+        headers: this.headers,
+      }).then(res => {
+      if (res.status !== 200) throw res;
```
Comment (MrJacz) [inline src/lib/Client.js]:
```js
if (!res.ok) throw res;
```

File: package.json [hunk 6]
```diff
@@ -22,7 +22,7 @@
   },
   "homepage": "https://github.com/YorkAARGH/Idiotic-Wrapper#readme",
   "dependencies": {
-    "snekfetch": "^3.6.4"
+    "node-fetch": "^2.1.2",
```
Comment (MrJacz) [inline package.json]: no comma

**Comment by YorkAARGH** [comment]
This has already been implemented in version 1.0.0

**Pull request closed without merging**